endoreg_db/services/pdf_import.py

```
@@ -573,7 +573,72 @@ def _parse_date_field(self, raw_value, meta_key, sm_field):
+            if 'anonymized' in self.current_pdf.__dict__:
+                update_fields.append('anonymized')
+            self.current_pdf.save(update_fields=update_fields)
+
+            # Mark state as anonymized immediately; this keeps downstream flows working
+            state = self._ensure_state(self.current_pdf)
+            if state and not state.anonymized:
+                state.mark_anonymized(save=True)
+
+            logger.info("Updated anonymized_file reference to: %s", self.current_pdf.anonymized_file.name)
+
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)